extensions/2.0/Khronos/KHR_texture_astc/README.md

```diff
@@ -0,0 +1,253 @@
+
+At runtime, engines can directly use the ktx images unlike EXT_texture_basisu which requires doing transcoding to other compressed formats. This also means you get support for all ASTC block formats and quality presets.
+
+## glTF Schema Updates
+
+The `KHR_texture_astc` extension is added to the `textures` object and specifies a `source` property that points to the index of the `image` which defines a reference to the KTX v2 image with ASTC compression.
+
+The following glTF will load `astc_image.ktx2` in clients that support this extension, and fall back to `image.png` otherwise.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.png"
+        },
+        {
+            "uri": "astc_image.ktx2"
+        }
+    ]
+}
+```
+
+When used in the glTF Binary (GLB) format the `image` that points to the KTX v2 resource uses the `mimeType` value of `image/ktx2`.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "mimeType": "image/png",
+            "bufferView": 1
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX v2 image with ASTC compression without a fallback, define `KHR_texture_astc` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "astc_image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_astc.schema.json](schema/texture.KHR_texture_astc.schema.json)
+
+## KTX v2 Images with Astc compression 
+
+To cover a broad range of use cases, this extension allows different ASTC block sizes as well as LDR, HDR and 3D images. These can be determined from the vkFormat of the KTX image.
```

Comment:
Given the fragmented nature of ASTC implementations you may need to have different extensions for LDR only, LDR+HDR and 3D.

Reply:
No sure what you mean by texture slots. If one reads the texture format and works out its an HDR image, the shader code for sampling that texture can be what's required. The hardware can deal with sampling this in any space.

You will always have special use of HDR images and special code/shader code to deal with it.

Reply:
The logical model of glTF goes like this:

1. An `image` is just a resource reference.
2. A `texture` is an approximation of a 2D GPU texture object so it refers to an `image` (called `source` for historical reasons) and a sampler state (wrapping and filtering modes). To ensure ecosystem consistency, unextended glTF textures may only refer to JPEG/PNG images. A texture extension usually provides an alternative representation of a JPEG/PNG image source therefore that alternative source must not contain anything that cannot be represented as a regular JPEG/PNG image.
3. Many material properties, e.g., PBR base color, are texturable. This means that a property like `baseColorTexture` points to a texture object that provides ready-to-use per-texel values for that property. All such properties expect texel values in the normalized `[0, 1]` range and they are not aware of any texture encoding details.

All of this means that a texture object that refers to an image with HDR values cannot be used for providing texel data for glTF material properties.

Reply:
Thanks for explaining this. 

> Many material properties, e.g., PBR base color, are texturable. 

I understand the glTF workflow and it makes perfect sense if you just think about using HDR images for PBR materials however many uses of HDR/float textures in graphics are not about storing color/PBR material properties like albedo/normal etc directly but rather things like environment maps or other image base lighting parameters. This is for that specific usecase.

> therefore that alternative source must not contain anything that cannot be represented as a regular JPEG/PNG image

And I don't think I am proposing that here. You can still have an alternative non-HDR JPEG/PNG image for your HDR texture target.

Reply:
Fixed in https://github.com/KhronosGroup/glTF/pull/2369/commits/8781d2296877799387f293c4ebc3da6bd0b9a254